pymove/visualization/folium.py

```diff
@@ -631,7 +633,12 @@ def _filter_and_generate_colors(
         else:
             colors = color[:]
         items = [*zip(ids, colors)]
-
+        if color_by_id is not None:
+            keys = color_by_id.keys()
+            for key in keys:
+                for i in range(len(items)):
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6225566)